src/Uno.UI/UI/Xaml/Markup/Reader/XamlObjectBuilder.cs

```diff
@@ -90,7 +98,14 @@ public XamlObjectBuilder(XamlFileDefinition xamlFileDefinition)
 
 			if (createInstanceFromXClass && TypeResolver.FindType(classMember?.Value?.ToString()) is { } classType)
 			{
-				return Activator.CreateInstance(classType);
+				var created = Activator.CreateInstance(classType);
+
+				if (created is FrameworkElement fe && !string.IsNullOrWhiteSpace(_fileUri))
```

Comment:
It's unlikely that it will be empty/whitespace. I guess `_fileUri is not null` should be sufficient.